packages/chain-adapters/src/solana/SolanaChainAdapter.ts

```diff
@@ -0,0 +1,354 @@
+
+      if (pubKey) return pubKey
+
+      if (!supportsSolana(wallet)) throw new Error('Wallet does not support Solana.')
+
+      const address = await wallet.solanaGetAddress({
+        addressNList: toAddressNList(this.getBIP44Params({ accountNumber })),
+        showDisplay: showOnDevice,
+      })
+
+      if (!address) throw new Error('Unable to generate Solana address.')
+
+      return address
+    } catch (err) {
+      return ErrorHandler(err)
+    }
+  }
+
+  async getAccount(pubkey: string): Promise<Account<KnownChainIds.SolanaMainnet>> {
+    try {
+      const data = await this.providers.http.getAccount({ pubkey })
+
+      const balance = BigInt(data.balance) + BigInt(data.unconfirmedBalance)
+
+      return {
+        balance: balance.toString(),
+        chainId: this.chainId,
+        assetId: this.assetId,
+        chain: this.getType(),
+        chainSpecific: {
+          tokens: data.tokens.map(token => ({
+            assetId: toAssetId({
+              chainId: this.chainId,
+              assetNamespace: 'spl',
+              assetReference: token.id,
+            }),
+            balance: token.balance,
+            name: token.name,
+            precision: token.decimals,
+            symbol: token.symbol,
+          })),
+        },
+        pubkey,
+      }
+    } catch (err) {
+      return ErrorHandler(err)
+    }
+  }
+
+  async getTxHistory(input: TxHistoryInput): Promise<TxHistoryResponse> {
+    const requestQueue = input.requestQueue ?? new PQueue()
+
+    try {
+      const data = await requestQueue.add(() =>
+        this.providers.http.getTxHistory({
+          pubkey: input.pubkey,
+          pageSize: input.pageSize,
+          cursor: input.cursor,
+        }),
+      )
+
+      const txs = await Promise.all(
+        data.txs.map(tx => requestQueue.add(() => this.parseTx(tx, input.pubkey))),
+      )
+
+      return {
+        cursor: data.cursor ?? '',
+        pubkey: input.pubkey,
+        transactions: txs,
+      }
+    } catch (err) {
+      return ErrorHandler(err)
+    }
+  }
+
+  async buildSendTransaction(input: BuildSendTxInput<KnownChainIds.SolanaMainnet>): Promise<{
+    txToSign: SignTx<KnownChainIds.SolanaMainnet>
+  }> {
+    try {
+      const { accountNumber, to, value, chainSpecific } = input
+
+      const { blockhash } = await this.connection.getLatestBlockhash()
+
+      const computeUnitLimit = chainSpecific.computeUnitLimit
+        ? Number(chainSpecific.computeUnitLimit)
+        : undefined
+
+      const computeUnitPrice = chainSpecific.computeUnitPrice
+        ? Number(chainSpecific.computeUnitPrice)
+        : undefined
+
+      const txToSign: SignTx<KnownChainIds.SolanaMainnet> = {
+        addressNList: toAddressNList(this.getBIP44Params({ accountNumber })),
+        blockHash: blockhash,
+        computeUnitLimit,
+        computeUnitPrice,
+        // TODO: handle extra instructions
+        instructions: undefined,
```

Comment:
We will need implement this very soon to support THORChain bridges as early as possible (even if it's the only usage I could detect despite DeFi products we would want to implement)

Reply:
Solana isn't supported yet on thorchain is it? Regardless, it will be implemented upon token support and then can follow suite with other defi interactions.

Reply:
Not supported yet but soon I guess

You're right we will need it to support tokens anyway as it's not a regular `send` transaction, so I think we will support it very soon naturally